### PR TITLE
Remove make routes

### PR DESCRIPTION
**Pull request by AboudyKreidieh**
changes:
- stopped adding vehicles to the network before the environment is created (in `make_routes`). Vehicles are now only imported in the environment's `reset` method, which is more compatible with aimsun. Now `generate_starting_positions` is only called within the environment.
- removed `vehicle_arrangement_shuffle` and `starting_position_shuffle`, and replaced them with `shuffle` from `InitialConfig` (which essentially exists to perform the same task)

Note: 
- some of the tests needed to be modified by calling `reset` before they start since there are no cars in the network before reset.

### CODE REVIEW

File: flow/envs/base_env.py [hunk 1]
```diff
@@ -496,7 +427,6 @@ def reset(self):
                 self.traci_connection.vehicle.unsubscribe(veh_id)
                 self.vehicles.remove(veh_id)
             except (FatalTraCIError, TraCIException):
-                print("Error during start: {}".format(traceback.format_exc()))
```
Comment (fywu85) [inline flow/envs/base_env.py]: Perhaps throw a warning before passing?

File: flow/envs/base_env.py [hunk 2]
```diff
@@ -507,7 +437,7 @@ def reset(self):
                 self.traci_connection.vehicle.remove(veh_id)
                 self.traci_connection.vehicle.unsubscribe(veh_id)
             except (FatalTraCIError, TraCIException):
-                print("Error during start: {}".format(traceback.format_exc()))
```
Comment (fywu85) [inline flow/envs/base_env.py]: Perhaps throw a warning before passing?

File: flow/envs/base_env.py [hunk 3]
```diff
@@ -550,6 +480,11 @@ def reset(self):
         # update the colors of vehicles
         self.update_vehicle_colors()
 
+        # check to make sure all vehicles have been spawned
+        if len(self.initial_ids) < self.vehicles.num_vehicles:
+            logging.error("Not enough vehicles have spawned! Bad start?")
+            sys.exit()
```
Comment (fywu85) [inline flow/envs/base_env.py]: For debugging purpose, perhaps print which vehicles were not spawned and why?

File: flow/envs/multiagent_env.py [hunk 4]
```diff
@@ -241,7 +210,7 @@ def reset(self):
                 self.traci_connection.vehicle.remove(veh_id)
                 self.traci_connection.vehicle.unsubscribe(veh_id)
             except (FatalTraCIError, TraCIException):
-                print("Error during start: {}".format(traceback.format_exc()))
```
Comment (fywu85) [inline flow/envs/multiagent_env.py]: Same here, why do we not want to throw a warning?

File: tests/fast_tests/test_environments.py [hunk 5]
```diff
@@ -688,14 +688,19 @@ def test_reset_inflows(self):
 
         env = DesiredVelocityEnv(env_params, sumo_params, scenario)
 
+        # reset the environment and get a new inflow rate
+        env.reset()
+        expected_inflow = 1353.6  # just from checking the new inflow
```
[the remark this answers is not in the document]
Comment (AboudyKreidieh) [reply tests/fast_tests/test_environments.py]: The explanation is as mentioned. I checked the new inflows (which are random, but seeded) and these were the new values.

File: tests/fast_tests/test_environments.py [hunk 6]
```diff
@@ -688,14 +688,19 @@ def test_reset_inflows(self):
...
         for _ in range(500):
             env.step(rl_actions=None)
-        self.assertAlmostEqual(env.vehicles.get_inflow_rate(250)/1500, 1, 2)
+        self.assertAlmostEqual(
+            env.vehicles.get_inflow_rate(250)/expected_inflow, 1, 2)
 
         # reset the environment and get a new inflow rate
         env.reset()
-        expected_inflow = 1353.6  # just from checking the new inflow
+        expected_inflow = 1756.8  # just from checking the new inflow
```
[the remark this answers is not in the document]
Comment (AboudyKreidieh) [reply tests/fast_tests/test_environments.py]: Same here.

File: flow/envs/base_env.py (hunk 2 above)
Comment (AboudyKreidieh) [reply flow/envs/base_env.py]: there was a little bug. fixed it and re-introduced the error message

File: flow/envs/multiagent_env.py (hunk 4 above)
Comment (AboudyKreidieh) [reply flow/envs/multiagent_env.py]: again, got rid of bug that necessitated this

File: flow/envs/base_env.py (hunk 3 above)
Comment (AboudyKreidieh) [reply flow/envs/base_env.py]: added information on the names of vehicles and their initial states for ease of debugging

File: flow/envs/base_env.py (hunk 1 above)
Comment (AboudyKreidieh) [reply flow/envs/base_env.py]: brought it back

**Comment by AboudyKreidieh** [comment]
@fywu85 I resolved all the issues u mentioned and need to move forward (since this is a relatively small PR), so merging this

**Pull request merged**